src/observability/publisher.go

```diff
@@ -1,38 +1,52 @@
 package observability
 
 import (
+	"sync"
+
+	cfg "github.com/accuknox/auto-policy-discovery/src/config"
 	"github.com/accuknox/auto-policy-discovery/src/types"
 )
 
 func ProcessSystemSummary() {
-	if len(PublisherMap) <= 0 {
+
+	if len(SummarizerMap) <= 0 {
```

Comment:
```suggestion
	if len(SummarizerMap) == 0 {
```
`len()` of `map` won't be a negative value.